plugins/guests/linux/cap/public_key.rb

```diff
@@ -53,6 +53,7 @@ def self.remove_public_key(machine, contents)
               result=$?
             fi
             rm -f '#{remote_path}'
+            chmod 0600 ~/.ssh/authorized_keys
```

Comment:
Hi briancain

I tried to check master branch, it but it seems to have not been fix yet. 
[here](https://github.com/mitchellh/vagrant/blob/master/plugins/guests/linux/cap/public_key.rb#L54)

I am sorry if I look at a different branch.